### ISSUE TITLE
conflicted file no longer marked conflicted after commit of other files

### ISSUE THREAD

**Issue by msftrncs** [desktop/desktop#6345]
## Description
In the case a merge/stash conflict is in the working area, committing other files loses the conflicted status of the conflicted files.

## Version
* GitHub Desktop: 1.5.1-beta1+ (Built on Node 10)
* Operating system: Win 10 1809

## Steps to Reproduce
1. Make changes to a couple files in a repository (in this case, Desktop)
1. decide to change branches, so stash changes (VS Code was used for this)
1. switch branches
1. apply latest stash (again, VS Code was used for this)
    1. in this case a conflict occurred in one file, VS Code had it marked 'C'
1. use Desktop to commit the other file, by unchecking the conflicted file, Desktop does this, VS Code will not.
1. at worst, modify an additional file and save it, then check status of the repository. (only because I had already done this by the time I noticed this issue)
1. verify conflicted file still has conflict markers.
    1. Desktop will not show the conflict markers as conflicts, instead showing them as changes. (see Add. Info.)
    1. VS Code shows the conflict markers, but has lost the action buttons to resolve the conflicts.

### Expected Behavior
I would have expected the conflicted file to remain conflicted.

### Actual Behavior
Apparently the file lost its conflicted status during the process to commit the other file, as was probably necessary.  

## Additional Information
Maybe this is the reason that VS Code refuses to let me commit other files while there is one marked conflicted.  I take it with my limited understanding of GIT that the conflicted file is so marked in an index, and everything in the index is in queue for a commit, and that is why the commit is blocked while a file is still marked in conflicted state.  I don't know if after Desktop removes the conflicted file from the index, and completes the commit, if it can put it back in the index in conflicted status.  From VS Code's SCP view, the conflicted file is marked as NOT staged.

IMHO I don't understand why a commit cannot complete WITHOUT the conflicted files, normally.  It would seem that making a commit without the conflicts would be a good thing, at least some times.

A view of what things looked like in Desktop:
![image](https://user-images.githubusercontent.com/26179051/49492000-16cb6000-f81c-11e8-85e7-e540a1c2e606.png)

So its understood what I was doing, I was working on multiple issues across several branches at one time in the working directory, and then trying to commit them each back to the correct branch.  In this case, the conflicted file is of no interest, I'll just un-stash it again in the correct branch, but not ready to commit it just yet.

**Comment by billygriffin** [comment]
@msftrncs Thanks for this report. I'm having a bit of trouble digesting this as it seems to get to some of the underlying technical components of merge conflicts that I don't have a super strong grasp of.

@tierninho Would you mind trying to reproduce and/or adding additional color here, and also cc: @outofambit as you've been working through the merge conflicts stuff.

**Comment by tierninho** [comment]
I gave this a try, but no luck on my end reproducing it as I feel the same way for this 👇 

>  I'm having a bit of trouble digesting this as it seems to get to some of the underlying technical components of merge conflicts that I don't have a super strong grasp of.

**Comment by msftrncs** [comment]
I've determined a fairly quick way to reproduce the issue.

1. starting with a fresh repository
1. create a file and commit
1. make a branch, modify the file, and commit
1. switch back to master, modify the file, stash
1. switch to the non-master branch, apply stash, a merge conflict will occur.

In VS Code, the conflicted file is marked C (conflicted) but in Desktop the file is just 'checked' and the diff view doesn't notice the conflict markers.

![image](https://user-images.githubusercontent.com/26179051/49917034-1f96e400-fe63-11e8-88b8-9ef29fd0d9ee.png)

6. create another file in the repository.

This file is marked 'U' (untracked) and is marked as added in Desktop.

7. unselect the conflicted file and commit the new file.

Now VS Code thinks the conflicted file is just modified, and the same in Desktop.

![image](https://user-images.githubusercontent.com/26179051/49917312-4bff3000-fe64-11e8-9468-ecc428a4eb3b.png)

### notes

- I was only modifying one line files, changing the text of the one line, a different short phrase or word each time, so that every single commit is different.
- commits were accomplished in Desktop.
- stash/apply were accomplished in VS Code.

This doesn't really help you understand why I am doing this.  I do this so I can build a project with several unrelated untested changes at the same time when I know they're otherwise not going to conflict with each other, but then when I commit them, they need to be in different branches (so they are in separate PR's) and those branches happen to be at slightly different points from master, so as I switch branches (stash, switch, apply) some of the files go in to conflict.  Because the stash contains all the changed files, I can just revert the conflicted file that is not part of the branch I am committing to, because when I switch to another branch I can apply the same stash back in again.   So its not super important, but as Desktop grows you will probably want to make sure you handle this situation gracefully.  Of course, VS Code just doesn't let you commit anything while a file is conflicted, so there is always that route.

**Comment by tierninho** [comment]
@msftrncs I can successfully reproduce this. Thanks for clarifying the steps!

So after step 5, I got:

```
$ git stash pop
Auto-merging one.txt
CONFLICT (content): Merge conflict in one.txt
```

and 

(file is conflicted)
<img width="1274" alt="screen shot 2018-12-13 at 3 56 47 pm" src="https://user-images.githubusercontent.com/14828183/49974784-ba3d0480-feef-11e8-8524-6b450df4c394.png">

After step 6 (file is still conflicted):
<img width="1275" alt="screen shot 2018-12-13 at 3 58 08 pm" src="https://user-images.githubusercontent.com/14828183/49974813-ece6fd00-feef-11e8-9119-12d8e0bc65c5.png">

After step 7 (file has changed to "modified"):
<img width="1279" alt="screen shot 2018-12-13 at 3 59 20 pm" src="https://user-images.githubusercontent.com/14828183/49974851-143dca00-fef0-11e8-972f-f1dadaf35606.png">

with 

```
$ git status
On branch dsfsdf
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   one.txt

no changes added to commit (use "git add" and/or "git commit -a")
```

@outofambit does this look like a bug to you?

**Comment by sergiou87** [comment]
As far as I can tell, this is mostly based on how git works and I experience the same behavior on VS Code. It's been 6 years since this was reported (sorry 😅 ) so maybe things have changed. In any case, closing this now…
